cmd/subctl/join.go

```diff
@@ -291,11 +291,10 @@ func determineClusterID(status reporter.Interface) {
 	}
 }
 
-func getNetworkDetails(clientProducer client.Producer, status reporter.Interface) *network.ClusterNetwork {
+func getNetworkDetails(controllerClient controllerclient.Client, status reporter.Interface) *network.ClusterNetwork {
 	status.Start("Discovering network details")
 
-	networkDetails, err := network.Discover(clientProducer.ForDynamic(), clientProducer.ForKubernetes(), clientProducer.ForOperator(),
-		constants.OperatorNamespace)
+	networkDetails, err := network.Discover(controllerClient, constants.OperatorNamespace)
```

Comment:
Likewise, the function declaration could be left unchanged and `clientProducer.ForGeneral()` used instead.

Reply:
See above comment.